src/scm/Model.ts

```diff
@@ -347,6 +320,43 @@ export class Model implements Disposable {
         });
 
         this._onDidChange.fire(this.ResourceGroups);
+    }
+
+    private async getDepotOpenedFilePaths(): Promise<string[]> {
+        const output = await Utils.getOutput('opened');
+        const opened = output.trim().split('\n');
+        if (opened.length === 0) {
+            return;
+        }
+
+        const files = [];
+        opened.forEach(open => {
+            const matches = open.match(/(.+)#(\d+)\s-\s([\w\/]+)\s(default\schange|change\s\d+)\s\((\w+)\)/);
+            if (matches) {
+                files.push(matches[1]);
+            }
+        });
+
+        return files;
+    }
 
+    private async getFstatInfoForFiles(files: string[]): Promise<any> {
+        const fstatOutput: string = await Utils.getOutput(`fstat ${files.join(' ')}`);
```

Comment:
This won't work for files with spaces in the path
(e.g. //depot/source/file with space.txt#1)